### PR TITLE
Feat: API endpoint update and tests

### PR DESCRIPTION
**Pull request by olivernybroe**
Sorry for the big commit, should prob. had split it up in multiple commits.

- Changed election to electors in API for more restful syntax
- Changed candidate to candidates in API
- Changed election to elections in API
- Added tests for candidates API
- Updated test to the new endpoints
- Refactored api route file 
- Fixed a bug which allowed to delete and access electors in other elections then the one you had selected in API
- Added support for updating a electors name through API  (does not update ballot_version)
- Some small refactoring

### CODE REVIEW

**Comment by olivernybroe** [comment]
Oh, forgot to update the front-end to use the new endpoints, forgot that the front-end uses the API for getting data.

**Comment by olivernybroe** [comment]
Should have updated all the endpoints now in front-end also. Not sure if I missed some, we really need tests for front-end also.

File: routes/api.php [hunk 1]
```diff
@@ -18,27 +19,32 @@
...
-Route::resource('election', 'ElectionController', ['only' => ['index', 'store', 'show', 'update', 'destroy']]);
-Route::post('election/{election_id}/batchvote', 'ElectionController@batchvote')->name('election.batchvote');
-Route::get('election/{election_id}/batchvote', 'ElectionController@batchvote_view')->name('election.batchvote_view');
-Route::get('election/{election_id}/get_ready', 'ElectionController@get_ready')->name('election.get_ready');
-Route::post('election/{election_id}/set_ready', 'ElectionController@set_ready')->name('election.set_ready');
-Route::get('election/{election_id}/voter_stats', 'ElectionController@voter_stats')->name('election.voter_stats');
-Route::get('election/{election_id}/voter_details', 'ElectionController@voter_details')->name('election.voter_details');
-Route::post('election/{election_id}/batch_candidates', 'ElectionController@batch_candidates')->name('election.batch_candidates');
-Route::get('election/{election_id}/batch_candidates', 'ElectionController@batch_candidates_view')->name('election.batch_candidates_view');
+Route::group(['middleware' => 'auth:api'], function (Router $api) {
```
Comment (carlschroedl) [inline routes/api.php]: Aha! I didn't know how to apply middleware to multiple routes at a time. Thanks for teaching me!
@neckberg @tylerharter if you want to learn more:
https://laravel.com/docs/5.6/routing#route-group-middleware

File: storage/api-docs/api-docs.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
     },
```
Comment (carlschroedl) [inline storage/api-docs/api-docs.json]: Is this file updated manually or automatically? If it is updated automatically, would it make sense to add it to the .gitignore?

File: tests/Feature/CandidateTest.php [hunk 3]
```diff
@@ -0,0 +1,554 @@
...
+use Carbon\Carbon;
+use Illuminate\Auth\Access\AuthorizationException;
+use Illuminate\Auth\AuthenticationException;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+use Illuminate\Support\Collection;
+use Illuminate\Validation\ValidationException;
+use Tests\TestCase;
+
+class CandidateTest extends TestCase
```
Comment (carlschroedl) [inline tests/Feature/CandidateTest.php]: Woohoo! So many tasty tests!

File: tests/TestCase.php [hunk 4]
```diff
@@ -26,4 +28,19 @@ public function render($request, \Exception $e)
             }
         });
     }
+
```
Comment (carlschroedl) [inline tests/TestCase.php]: Cool! These utility assertions will be very helpful.

File: storage/api-docs/api-docs.json (hunk 2 above)
Comment (olivernybroe) [reply storage/api-docs/api-docs.json]: Think it's the auto updated one. You are right, it should be ignored.

**Comment by carlschroedl** [comment]
> we really need tests for front-end also.
Very true!

**Comment by neckberg** [comment]
Changes look good to me, and I think changing the resource names to plural makes sense / is in line with common convention. I'm going to go ahead and merge, since Carl has already given his approval.

**Pull request merged**